chess/frontend/gui/src/modules/Game/Game.jsx

```diff
@@ -163,55 +176,88 @@ export default class Game extends React.Component {
       method: "GET",
       data: { index: TRANSLATE_POSITION[index] }
     }).then(response => {
-      this.highlightMoves(response.moves)
+      this.highlightMoves(response.moves);
     });
   }
 
+  // highlights the squares on the board that are 'movable' according
+  // to the response from the server
   highlightMoves(moves) {
     let newSquares = this.state.squares;
 
     moves.forEach(move => {
-      newSquares[TRANSLATE_POSITION[move]].canMovePiece()
+      newSquares[TRANSLATE_POSITION[move]].canMovePiece();
     });
 
     this.setState({
       squares: newSquares
     });
   }
 
+  // unhighlights the 'movable' pieces on the board
+  dehighlightMoves() {
+    let newStateSquares = this.state.squares;
+    newStateSquares.forEach(square => {
+      if (square.movable) {
+        square.cannotMovePiece();
+      }
+    });
+
+    this.setState({
+      squares: newStateSquares
+    });
+  }
+
   // Function that is passed down to the child components used
   // to handle moving the pieces
   handleClick(index) {
-    this.checkPossibleMoves(index);
-    // If the user selects the oponent's peice first
-    if (
-      this.state.selected === NOT_SELECTED && this.state.turn !== this.state.squares[index].player) {
-      this.setState({
-        error: "Cannot move opponent's piece"
-      });
-    } else if (this.state.selected !== NOT_SELECTED) {
-      let captured = this.state.squares[index].player === this.opponentColor()
-      this.handleMove(captured, index);
-    }
-
-    // If the user selects a spot on the board that has a piece
-    else if (this.state.squares[index] != null) {
-      let newState = this.state.squares;
-      // Check whether or not that peice is already selected
-      if (this.state.squares[index].selected) {
-        newState[index].deselectPiece();
-        this.setState({
-          squares: newState,
-          selected: NOT_SELECTED,
-          error: ""
-        });
-      } else {
+    let newState = this.state.squares;
+    // if the player currently has no piece selected
+    if (this.state.selected === NOT_SELECTED) {
+      // if the spot that was clicked was the current player's color
+      if (this.state.squares[index].player === this.state.turn) {
```

Comment:
Maybe this check can be wrapped in a function, `checkIfPieceBelongsToCurrentPlayer()`? A mouthful, but gets the logic across more easily then then `this.state.squares[index].player === this.state.turn`